torch/distributed/_tensor/_dispatch.py

```diff
@@ -231,6 +237,19 @@ def default_tensor(spec: DTensorSpec) -> torch.Tensor:
             if output_sharding.output_spec is not None:
                 return args[0]
             else:
+                if op_call == aten._amp_foreach_non_finite_check_and_unscale_.default:
```

Comment:
I think logic like this kind of special handling should be implemented as a custom op handler https://github.com/pytorch/pytorch/blob/main/torch/distributed/_tensor/_dispatch.py#L97, instead of hard coding it inside the dispatching logic

Reply:
Switched to `self._custom_op_handlers`